### PR TITLE
026 branding

### PR DESCRIPTION
**Pull request by johnhutchman**

### CODE REVIEW

**Review by niall-mccloy — APPROVED** [review]

File: views/home.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    <title>Kainos Academy</title>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css">
+    <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="../public/assets/style/styles.css">
+    <link rel="icon" type="image/x-icon" href="https://companieslogo.com/img/orig/KNOS.L-8a0d29cc.png?t=1720244492">
+
+</head>
+<body>
+
+    <special-header></special-header>
```
Comment (mlynch147) [inline views/home.html]: more of a question than anything.... can you use a self closing tag here.... eg `<special-header />`

File: views/headerfooter.js
Comment (mlynch147) [inline views/headerfooter.js:1]: @johnhutchman couple of points from me
- can you split this up into two files... one for the header and one for the footer.  
- can you put them in a new folder inside 'views' called 'components' or 'partials' or something similar...   
- rename the elements - MyHeader and MyFooter isnt really great... maybe KainosHeader, KainosFooter
- change the 'define' names to kainos-header and kainos-footer (not special)

File: views/home.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                        Software Engineer</a>
+                    <a href="https://careers.kainos.com/gb/en/job/JR_14214/Sales-Engineer" target="_blank"
+                        rel="noopener noreferrer" class="list-group-item list-group-item-action">Sales Engineer</a>
+                        
+                </div>
+            </section>
+            <br>
+            <br>
+        </main>
+        <special-footer></special-footer>
```
Comment (mlynch147) [inline views/home.html]: same as above - can you use a self closing tag ????

File: views/headerfooter.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+    <p class="mb-0">&copy; Kainos 2024</p>
+</footer>`;
+    }
+}
+
+
+customElements.define('special-header', MyHeader);
+
+customElements.define('special-footer', MyFooter);
```
Comment (TomBillingtonUK) [inline views/headerfooter.js]: I understand what you are trying to do: break the code into components and use them.

You aren't using the features of nunjucks though and are using the HTML 5 way of doing things.

What I would expect is something like this:


**views/components/footer.njk**
```
<footer>
...code here
</footer>
```

views/index.njk
```
... code here
{% include "components/footer.njk" %}
... code here
```

On top of that you could also use a layout and do something like this:

/views/layout.njk
```
<HTML>
   <body>
    {% include "components/header.njk" %}

    {% block content %}
    {% endblock %}

    {% include "components/footer.njk" %}
  </body>
</html>
```

And then your index.njk file becomes

```
{% extends "layout.njk" %}
{% block content %}
... index page content
{% endblock %}
```

So when you go and create multiple pages you don't need to copy and paste all the boiler plate

File: views/home.html (hunk 1 above)
Comment (johnhutchman) [reply views/home.html]: Cant use closing tags on HTMLElements

File: views/headerfooter.js
Comment (johnhutchman) [reply views/headerfooter.js:1]: all changes done

**Review by mlynch147 — APPROVED** [review]

**Pull request merged**